### PR TITLE
Smaller timeouts for broker reconnect

### PR DESCRIPTION
**Pull request by soffokl**

### CODE REVIEW

File: services/wireguard/service/service_unix.go [hunk 1]
```diff
@@ -112,7 +107,7 @@ func (manager *Manager) ProvideConfig(publicKey json.RawMessage) (session.Servic
 		}
 	}
 
-	return config, manager.once(destroy), nil
+	return config, destroy, nil
```
Comment (tadovas) [inline services/wireguard/service/service_unix.go]: This seems to be unrelated to PR

File: communication/nats/sender.go [hunk 2]
```diff
@@ -94,3 +95,12 @@ func (sender *senderNATS) Request(producer communication.RequestProducer) (respo
 
 	return responsePtr, nil
 }
+
+func checkConnection(conn Connection) {
```
Comment (tadovas) [inline communication/nats/sender.go]: I would like to see this function part of connection. I.e. conn.Check

File: communication/nats/discovery/config.go [hunk 3]
```diff
@@ -23,6 +23,6 @@ import "time"
 const (
 	BrokerPort          = 4222
 	BrokerMaxReconnect  = -1
-	BrokerReconnectWait = 4 * time.Second
+	BrokerReconnectWait = 1 * time.Second
```
Comment (tadovas) [inline communication/nats/discovery/config.go]: It would be nice all these tweaks to be configurable from cmd line, with sensible defaults

Comment (soffokl) [reply communication/nats/discovery/config.go]: The issue for that added: https://github.com/mysteriumnetwork/node/issues/818

File: services/wireguard/service/service_unix.go (hunk 1 above)
Comment (soffokl) [reply services/wireguard/service/service_unix.go]: Well, this is revert of the hack that was added because session destroys never worked. This PR fix session destroys, so these changes can be reverted. 
I can do it in another PR, but why not to include it here? :)

Comment (tadovas) [reply services/wireguard/service/service_unix.go]: Didn't notice the relation. All good 👍

File: communication/nats/discovery/config.go (hunk 3 above)
Comment (tadovas) [reply communication/nats/discovery/config.go]: 👍

File: communication/nats/sender.go (hunk 2 above)
Comment (soffokl) [reply communication/nats/sender.go]: Changed it to be part of the connection.

File: communication/nats/discovery/address.go [hunk 4]
```diff
@@ -136,3 +144,14 @@ func (address *AddressNATS) GetContact() market.Contact {
 		},
 	}
 }
+
+type connection struct {
+	*nats_lib.Conn
```
Comment (tadovas) [inline communication/nats/discovery/address.go]: 💯 points for delegate pattern
